swiss_locator/core/profiles/profile_results.py

```diff
@@ -86,6 +88,29 @@ def zRange(self):
     def type(self):
         return "swiss-profile-web-service"
 
+    def snapPoint(self, point, context):
+        result = QgsProfileSnapResult()
+
+        prev_distance = float('inf')
+        prev_elevation = 0
+        for k, v in self.distance_to_height.items():
+            # find segment which corresponds to the given distance along curve
+            if k != 0 and prev_distance <= point.distance() <= k:
+                dx = k - prev_distance
+                dy = v - prev_elevation
+                snapped_z = (dy / dx) * (point.distance() - prev_distance) + prev_elevation
+
+                if abs(point.elevation() - snapped_z) > context.maximumSurfaceElevationDelta:
+                    return QgsProfileSnapResult()
```

Comment:
not exactly sure, is it ok to exit the loop here, no chance to get a good result at a next iteration?

Reply:
Actually, I borrowed the code from the current QGIS profile implementation, but giving it a deeper read, I see that we can only reach that `if` clause (line 103) if we're already in the good profile segment (i.e., see line 98's condition), so if we don't return immediately, we'll be accepting the previous `if` condition (line 98) evaluation for all the other segments where the initial point distance value is not included. All other segments would be rejected as the condition `prev_distance <= point.distance() <= k` won't be met. So, returning in line 104 is allowing us to avoid line 98's `if` condition being evaluated repeatedly without never really entering it again.